### ISSUE TITLE
[1.12.2; Forge 2705] Custom mobs won't spawn 

### ISSUE THREAD

**Issue by diegtiarov** [McJty/InControl#66]
I've encountered a problem where Vanilla mobs keep spawning, whereas custom ones seem to be ignored.

Attaching files in .txt format (keep in mind only a few sections are entirely complete on behalf of testing; only required mobs are enabled)

[potentialspawn.txt](https://github.com/McJtyMods/InControl/files/2116101/potentialspawn.txt)
[spawn.txt](https://github.com/McJtyMods/InControl/files/2116102/spawn.txt)

Files have been checked for syntax as well as other errors.

The most interesting part of all this is that isolated cases will work (custom mobs will spawn when there are only a few rules listed in potentialspawn.json), for example:
```
{
        "dimension": 0,


        "biome": "Bayou",
        "mobs": [
            {
                "mob": "mocreatures:komododragon",
                "groupcountmin": 1,
                "groupcountmax": 3,
                "weight": 15
            }
        ]
    }
```

```
{
		"onjoin": true,
		"mob": "mocreatures:komododragon",
		"result": "deny"
	}
```
^It will work when there is only this in the files, plus all other spawns disabled.

Moreover, mobs will freely spawn when let loose (Vanilla spawner). I want them in specific biomes and count, though.

That's why I continued adding further sections, until now (I'm aware that most of the spawns are denied). When I tested the configs, however, only Vanilla mobs would spawn. That's quite disturbing because spawns on custom biomes work (biome names are correct), yet those are only Vanilla mobs which spawn.

So far, I've been clueless in identifying the cause of the problem. The only lead is that it may be caused by lists of biomes rather than each one typed in separately.

**Comment by McJty** [comment]
"biome": ["Birch Forest", "Birch Forest Hills", "Birch Forest Hills M", "Birch Forest M", "Boreal Forest", "Bog", "Coniferous Forest", "Fen", "Flower Forest", "Forest", "ForestHills", "Grove", "Maple Woods", "Mega Spruce Taiga", "Mega Taiga", "Mega Taiga Hills", "Overgrown Cliffs", "Redwood Forest", "Redwood Taiga Hills M", "Roofed Forest", "Roofed Forest M", "Sacred Springs", "Seasonal Forest", "Shield", "Snowy Coniferous Forest", "Snowy Forest", "Taiga", "Taiga M", "TaigaHills", "Woodland"],

I don't think those are the correct biome names. For example the internal name of Birch Forest is 'birch_forest' and those are the names that In Control recognizes

**Comment by diegtiarov** [comment]
Might that be the cause? I saw example stating "Plains" on the wiki, so I thought I have to use names hidden behind IDs, other problems around the internet somewhat confirm that using names such as minecraft:plains won't do the trick... unless they work in most recent version.

I will check those (both with 'modtag:biomename' and 'biomename') in no time and respond.

**Comment by diegtiarov** [comment]
Here are the results. Turns out I was right regarding biome names.

[results.txt](https://github.com/McJtyMods/InControl/files/2116511/results.txt)


EDIT: I will check all biome names again to assure if it's not them, perhaps even today.
What I also noticed with initial configs I've attached is that I still can manipulate what vanilla mobs spawn on certain biomes and in what count they can spawn (observed on Biomes O' Plenty biome called Wasteland).
